src/main/java/br/unicamp/cst/io/rest/RESTServer.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package br.unicamp.cst.io.rest;
+
+import br.unicamp.cst.core.entities.Memory;
+import br.unicamp.cst.core.entities.MemoryContainer;
+import br.unicamp.cst.core.entities.MemoryObject;
+import br.unicamp.cst.core.entities.Mind;
+import br.unicamp.cst.support.InterfaceAdapter;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import express.Express;
+import express.middleware.CorsOptions;
+import express.middleware.Middleware;
+
+/**
+ *
+ * @author rgudwin
+ */
+public class RESTServer {
+    
+    public RESTServer(Mind m, int port) {
+        this(m,port,false);
+    }
+    
+    public RESTServer(Mind m, int port, boolean pretty) {
+        Express app = new Express();
+        Gson gson;
+        if (pretty)
+            gson = new GsonBuilder().registerTypeAdapter(Memory.class, new InterfaceAdapter<MemoryObject>())
+                             .registerTypeAdapter(Memory.class, new InterfaceAdapter<MemoryContainer>())
+                             .setPrettyPrinting().create();
+        else 
+            gson = new Gson();
+        CorsOptions corsOptions = new CorsOptions();
+        corsOptions.setOrigin("*");
```

Comment:
To set CORS origin as `*` can be a massive security flaw in some cases. Are we mindful of what we are doing here?

Reply:
Changed to allow arbitrary patterns as origins, for clients using the server.